models/articles.model.js

```diff
@@ -41,3 +39,51 @@ exports.updateArticleVotes = (article_id, votes = 0) => {
       }
     });
 };
+
+exports.selectArticles = topicFilter => {
+  if (topicFilter) {
+    return db
+      .query(
+        `SELECT articles.article_id, title, topic, articles.author, articles.body, articles.created_at, articles.votes, COUNT(comment_id) ::INT AS comment_count
+      FROM articles
+          LEFT JOIN comments ON articles.article_id = comments.article_id
+      WHERE topic = $1
+      GROUP BY articles.article_id
+      ORDER BY articles.created_at`,
+        [topicFilter]
+      )
+      .then(({ rows }) => {
+        return rows;
+      });
+  } else {
+    return db
+      .query(
+        `SELECT articles.article_id, title, topic, articles.author, articles.body, articles.created_at, articles.votes, COUNT(comment_id) ::INT AS comment_count
+    FROM articles
+        LEFT JOIN comments ON articles.article_id = comments.article_id
+    GROUP BY articles.article_id
+    ORDER BY articles.created_at`
+      )
+      .then(({ rows }) => {
+        return rows;
+      });
+  }
+};
+
+exports.selectCommentsByArticleId = article_id => {
+  return db
+    .query(
+      `SELECT comment_id, votes, created_at, author, body FROM comments WHERE article_id = $1 ORDER BY created_at DESC`,
+      [article_id]
+    )
+    .then(({ rows }) => {
+      if (rows.length > 0) {
+        return rows;
+      } else {
+        return Promise.reject({
```

Comment:
We should be checking here if the article exists. There could be a article in our database that has no comments, which would return an empty array on our rows property. If an emty array comes back, we should check if the article exists. If it does, this would be a 200 and a perfectly valid request. If the article doesn't exist, ten it wouldbe a 404 :)